include/xtensor/xscalar.hpp

```diff
@@ -973,13 +974,21 @@ namespace xt
     template <bool is_const, class CT>
     inline void xscalar_stepper<is_const, CT>::to_begin() noexcept
     {
-        p_c = p_c->stepper_begin(p_c->shape()).p_c;
+        if (m_end)
+        {
+            p_c = p_c->stepper_begin(p_c->shape()).p_c;
```

Comment:
couldn't we simplify this to `p_c = p_c - 1;` ?

Reply:
The fact we choose the end to be `p_c + 1` is an implementation detail, so I don't want to propagate it outside.